src/linker/implementation.py

```diff
@@ -125,3 +76,15 @@ def _validate_container_exists(self, logs: List[Optional[str]]) -> List[Optional
         ):
             logs.append(err_str)
         return logs
+
+    @property
+    def validation_filename(self) -> str:
+        return self.name + "_validator"
+
+    @property
+    def singularity_image_path(self) -> str:
+        return self._get_container_full_stem + ".sif"
```

Comment:
Not part of this PR, but I should have named this helper function `_get_image_full_stem`. 

That said, should we just get rid of that altogether? The reason for it was that back then we didn't know whether the image was going to be a singularity or docker image, but now we only use singularity. Thought I guess we're talking about supporting docker in the future so idk.

Reply:
I think apptainer will let you use either a docker or singularity image; In that case it probably makes the most sense to just have the user put in the full image file path with extension. I can verify this works as expected in the containerization ticket and remove this property then

Reply:
OK, so you can _either_ specify a local singularity image _or_ a docker or singularity URL, but seemingly not a docker tar.gz, which i guess makes sense. I think it's best to take this out and plan to allow docker URLs later